#! /bin/sh /usr/share/dpatch/dpatch-run

## 02_manpage.dpatch
## by Jonas Meurer <mejo@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Improve the upstream manpage

@DPATCH@

--- cryptsetup-1.0.5/man/cryptsetup.8
+++ cryptsetup-1.0.5/man/cryptsetup.8
@@ -88,7 +88,9 @@
 query for passwords twice. Useful, when creating a (regular) mapping for the first time, or when running \fIluksFormat\fR.
 .TP
 .B "\-\-key-file, \-d"
-use file as key material. With LUKS, key material supplied in key files via \-d are always used for existing passphrases. If you want to set a new key via a key file, you have to use a positional arg to \fIluksFormat\fR or \fIluksAddKey\fR. If the key file is "-", stdin will be used.
+use file as key material. With LUKS, key material supplied in key files via \-d are always used for existing passphrases. If you want to set a new key via a key file, you have to use a positional arg to \fIluksFormat\fR or \fIluksAddKey\fR.
+
+If the key file is "-", stdin will be used. This is different from how cryptsetup usually reads from stdin. See section \fBNOTES ON PASSWORD PROCESSING\fR for more information.
 .TP
 .B "\-\-key-size, \-s"
 set key size in bits. Has to be a multiple of 8 bits. The key size is limited by the used cipher. See output of /proc/crypto for more information. Can be used for \fIcreate\fR or \fIluksFormat\fR, all other LUKS actions will ignore this flag, as the key-size is specified by the partition header. Default is 128.
@@ -112,7 +114,7 @@
 Do not ask for confirmation. This option is only relevant for \fIluksFormat\fR.
 .TP
 .B "\-\-timeout, \-t"
-The number of seconds to wait before timeout. This option is relevant evertime a password is asked, like \fIcreate\fR, \fIluksOpen\fR, \fIluksFormat\fR or \fIluksAddKey\fR.
+The number of seconds to wait before timeout. This option is relevant evertime a password is asked, like \fIcreate\fR, \fIluksOpen\fR, \fIluksFormat\fR or \fIluksAddKey\fR. It has no effect if used in conjunction with \-\-key-file.
 .TP
 .B "\-\-tries, \-T"
 How often the input of the passphrase shall be retried. This option is relevant evertime a password is asked, like \fIcreate\fR, \fIluksOpen\fR, \fIluksFormat\fR or \fIluksAddKey\fR. The default is 3 tries.
@@ -125,14 +127,16 @@
 .B "\-\-version"
 Show the version.
 
-.SH NOTES ON PASSWORD PROCESSING FOR REGULAR MAPPINGS
-\fIFrom a file descriptor or a terminal\fR: Password processing is new-line sensitive, meaning the reading will stop after encountering \\n. It will processed the read material with the default hash or the hash given by \-\-hash. After hashing it will be cropped to the key size given by \-s (default 256 bits).
+.SH NOTES ON PASSWORD PROCESSING
+\fIFrom a file descriptor or a terminal\fR: Password processing is new-line sensitive, meaning the reading will stop after encountering \\n. It will processed the read material (without newline) with the default hash or the hash given by \-\-hash. After hashing it will be cropped to the key size given by \-s (default 256 bits).
 
-\fIFrom stdin\fR: Reading will continue until EOF (so using e.g. /dev/random as stdin will not work). After that the read data will be hashed with the default hash or the hash given by \-\-hash and the result will be cropped to the keysize given by \-s (default 256 bits). If "plain" is used as an argument to the hash option, the input data will not be hashed. Instead it will be zero padded (if shorter than the keysize) or truncated (if longer than the keysize) and used directly as the key. No warning will be given if the amount of data read from stdin is less than the keysize.
+\fIFrom stdin\fR: Reading will continue until EOF (so using e.g. /dev/random as stdin will not work), with the trailing newline stripped. After that the read data will be hashed with the default hash or the hash given by \-\-hash and the result will be cropped to the keysize given by \-s (default 256 bits). If "plain" is used as an argument to the hash option, the input data will not be hashed. Instead it will be zero padded (if shorter than the keysize) or truncated (if longer than the keysize) and used directly as the key. No warning will be given if the amount of data read from stdin is less than the keysize.
 
 \fIFrom a key file\fR: It will be cropped to the size given by \-s. If there is insufficient key material in the key file, cryptsetup will quit with an error.
+
+If \-\-key-file=- is used for reading the key from stdin, no trailing newline ist stripped from the input. Without that option, cryptsetup strips trailing newlines from stdin input.
 .SH NOTES ON PASSWORD PROCESSING FOR LUKS
-Password processing is totally different for LUKS. LUKS uses PBKDF2 to protect against dictionary attacks (see RFC 2898). 
+LUKS uses PBKDF2 to protect against dictionary attacks (see RFC 2898). 
 LUKS will always use SHA1 in HMAC mode, and no other mode is supported at the moment. 
 Hence, \-h is ignored.
 
@@ -149,9 +153,9 @@
 .br
 LUKS extensions, and man page by Clemens Fruhwirth <clemens@endorphin.org>
 .SH "COMPATABILITY WITH OLD SUSE TWOFISH PARTITIONS"
-To read images created with SuSE Linux 9.2's loop_fish2 use --cipher
-twofish-cbc-null -s 256 -h sha512, for images created with even
-older SuSE Linux use --cipher twofish-cbc-null -s 192 -h
+To read images created with SuSE Linux 9.2's loop_fish2 use \-\-cipher
+twofish-cbc-null \-s 256 \-h sha512, for images created with even
+older SuSE Linux use \-\-cipher twofish-cbc-null \-s 192 \-h
 ripemd160:20
 .SH "REPORTING BUGS"
 Report bugs to <dm-crypt@saout.de>.
